### PR TITLE
[GLT-3286] workset search

### PR DESCRIPTION
**Pull request by avarsava**
Getting 'Library Aliquot cannot be filtered by workset.' on Workset 138 despite all the code being the same, don't understand it

### CODE REVIEW

**Comment by djcooke** [comment]
The `jQueryBackend` in `LibraryAliquotRestController` uses `PoolableElementViewService` rather than `LibraryAliquotService`. The `PaginationFilter` needs to go in `HibernatePoolableElementViewDao` instead of `HibernateLibraryAliquotDao`.

`PoolableElementView` is an alternate view of `LibraryAliquot` that has a more shallow graph and is used to improve performance in some places

**Comment by avarsava** [comment]
> The `jQueryBackend` in `LibraryAliquotRestController` uses `PoolableElementViewService` rather than `LibraryAliquotService`. The `PaginationFilter` needs to go in `HibernatePoolableElementViewDao` instead of `HibernateLibraryAliquotDao`.
> 
> `PoolableElementView` is an alternate view of `LibraryAliquot` that has a more shallow graph and is used to improve performance in some places

Done, thanks!

File: miso-web/src/main/webapp/scripts/urls.js [hunk 1]
```diff
@@ -376,7 +376,10 @@ Urls = (function() {
     bulkDelete: libraryRestBase + '/bulk-delete',
     spreadsheet: libraryRestBase + '/spreadsheet',
     bulkSave: libraryRestBase + '/bulk',
-    bulkSaveProgress: idUrlFunction(libraryRestBase + '/bulk')
+    bulkSaveProgress: idUrlFunction(libraryRestBase + '/bulk'),
+    worksetDatatable: function(worksetId) {
+      return libraryRestBase + '/dt/workset/' + worksetId;
+    },
```
Comment (djcooke) [inline miso-web/src/main/webapp/scripts/urls.js:382]: Can shorten to `idUrlFunction(libraryRestBase + '/dt/workset')` (and same for library aliquots and samples below)

File: sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateLibraryAliquotDao.java [hunk 2]
```diff
@@ -18,6 +18,7 @@
 import org.springframework.stereotype.Repository;
 import org.springframework.transaction.annotation.Transactional;
 
+import uk.ac.bbsrc.tgac.miso.core.data.Workset;
```
Comment (djcooke) [inline sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateLibraryAliquotDao.java:21]: Added unused import

File: miso-web/src/main/webapp/scripts/urls.js (hunk 1 above)
Comment (avarsava) [reply miso-web/src/main/webapp/scripts/urls.js:382]: I'm not sure I understand, these URLs need the workset ID, not the sample/library/libraryaliquot ID. will idUrlFunction still work? I'm not sure I understand where it gets 'id' from

Comment (djcooke) [reply miso-web/src/main/webapp/scripts/urls.js:382]: `idUrlFunction` returns a function that takes the workset ID as a parameter and appends it to the provided string (with a slash in-between), so it has the exact same effect as the code you wrote. It's defined at the top of `urls.js`. The same pattern is repeated many times, so these convenience functions keep the script much more tidy.

Comment (djcooke) [reply miso-web/src/main/webapp/scripts/urls.js:382]: (`idUrlFunction` doesn't specify or care that the parameter of the returned function is a workset ID. That's just the intention this case)

File: sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateLibraryAliquotDao.java (hunk 2 above)
Comment (avarsava) [reply sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateLibraryAliquotDao.java]: Fixed

File: miso-web/src/main/webapp/scripts/urls.js (hunk 1 above)
Comment (avarsava) [reply miso-web/src/main/webapp/scripts/urls.js]: Fixed

**Pull request merged**